### ISSUE TITLE
Add a newbreak package

### ISSUE THREAD

**Issue by simoncozens** [sile-typesetter/sile#802]
This could either be a Lua port or the nascent C port. Either one works.

**Comment by alerque** [comment]
Are you talking about [this JS code](https://github.com/simoncozens/newbreak)?

**Comment by simoncozens** [comment]
Yes. It's a much more maintainable justification engine and it supports alternates much *much* better than the current bag-on-the-side of Knuth-Plass. I'm not suggesting this go in as the main engine (Yet...) but it should be available as an add-on package.

**Comment by Omikhleia** [comment]
This issue did not received any activity for several years (and the aforementioned (Typescript) module  seems equally inactive for the past 4-years.)
I am closing it, but please feel free to re-open a dedicated issue if a different approach should be discussed.

Aside note: an add-on package is indeed perfectly doable independently from the core SILE distribution (e.g. as a 3rd-party package published on luarocks -- a solution that was not available when this issue was opened.)
